packages/maplibre/lib/map/apply-context-diff.ts

```diff
@@ -0,0 +1,61 @@
+import { MapContextDiff } from "@geospatial-sdk/core";
+import { Map } from "maplibre-gl";
+import { createLayer } from "./create-map";
+import { removeLayersFromSource } from "../helpers/map.helpers";
+
+/**
+ * Apply a context diff to an MapLibre map
+ * @param map
+ * @param contextDiff
+ */
+export async function applyContextDiffToMap(
+  map: Map,
+  contextDiff: MapContextDiff,
+): Promise<void> {
+  // removed layers (sorted by descending position)
+  if (contextDiff.layersRemoved.length > 0) {
+    const removed = contextDiff.layersRemoved.sort(
+      (a, b) => b.position - a.position,
+    );
+    for (const layerRemoved of removed) {
+      const maplibreLayer = map.getStyle().layers[layerRemoved.position];
+      if (maplibreLayer.type === "background") {
+        console.warn(
+          "[Warning] applyContextDiffToMap: removing background layer is not supported.",
+        );
+        continue;
+      }
+      const sourceId = maplibreLayer.source;
+      map.removeLayer(maplibreLayer.id);
+      map.removeSource(sourceId);
```

Comment:
I don't think this can work since there might be multiple maplibre layers per context layer. I guess we need to track maplibre layers by either the contextLayer id, or a hash of it. See https://github.com/camptocamp/geospatial-sdk/blob/fe1fda8882b1113ac9a52059a992836dd87cf23e/packages/core/lib/utils/hash.ts#L5